src/Raven.Server/Documents/Handlers/QueriesHandler.cs

```diff
@@ -123,17 +123,25 @@ public class QueriesHandler : DatabaseRequestHandler
             }
         }
 
-        private async Task Query(DocumentsOperationContext context, string indexName, OperationCancelToken token)
+        private async Task Query(DocumentsOperationContext context, string indexName, OperationCancelToken token, HttpMethod method)
         {
-            var query = IndexQueryServerSide.Create(HttpContext, GetStart(), GetPageSize(Database.Configuration.Core.MaxPageSize), context);
+            var indexQuery = IndexQueryServerSide.Create(HttpContext, GetStart(), GetPageSize(Database.Configuration.Core.MaxPageSize), context);
 
+            if (method == HttpMethod.Post && string.IsNullOrWhiteSpace(indexQuery.Query))
+            {
+                string queryString;
+                var request = context.Read(RequestBodyStream(), "QueryInPostBody");
+                if (request.TryGet("Query", out queryString))
```

Comment:
This should probably error if not there. If the req is a POST, we require a value.